### PR TITLE
Enablers for NFC-DEP support

### PR DESCRIPTION
**Pull request by monich**

### CODE REVIEW

File: src/nci_util.c [hunk 1]
```diff
@@ -159,10 +159,47 @@ nci_parse_mode_param(
...
+         *
+         * +=========================================================+
+         * | Offset | Size | Description                             |
+         * +=========================================================+
+         * | 0      | 1    | Bit Rate (1 = 212 kbps, 2 = 424 kbps)   |
+         * | 1      | 1    | SENSB_REF Response Length n (16 or 18)  |
+         * | 2      | n    | Bytes 2-17 of SENSF_RES                 |
+         * +=========================================================+
+         */
+        if (len > 1 && bytes[1] >= 8 && (bytes[1] + 2) <= len) {
```
Comment (spiiroin) [inline src/nci_util.c:176]: Random thought: n>=8 tests that about to be accessed amount is available. Should there be also record sanity check vs sizes 16/18? maybe for diagnostic purposes only?

File: src/nci_util.c [hunk 2]
```diff
@@ -308,6 +308,140 @@ nci_parse_iso_dep_poll_a_param(
...
+     * +=========================================================+
+     */
+    if (atr_res_len >= 15 && len >= atr_res_len + 1) {
+        const guint8* atr_res = bytes + 1;
+
+        /*
+         * NFCForum-TS-DigitalProtocol-1.0
+         * 14.6.3 ATR_RES Response
+         */
+        memcpy(param->nfcid3, atr_res, sizeof(param->nfcid3));
```
Comment (spiiroin) [inline src/nci_util.c:376]: nit: sizeof is not a function and param->nfcid3 is not a type -> superfluous parentheses.

File: src/nci_util.c [hunk 3]
```diff
@@ -308,6 +308,140 @@ nci_parse_iso_dep_poll_a_param(
...
+        param->did = atr_res[10];
+        param->bs = atr_res[11];
+        param->br = atr_res[12];
+        param->to = atr_res[13];
+        param->pp = atr_res[14];
+        if (atr_res_len > 15) {
+            param->g.bytes = atr_res + 15;
+            param->g.size = atr_res_len - 15;
```
Comment (spiiroin) [inline src/nci_util.c:384]: Funtion to "parse from bytes" which then caches pointer to 'bytes' data -- is it ok / going to stay valid?

Comment (spiiroin) [reply src/nci_util.c:384]: elsewhere too e.g. in nci_parse_nfc_dep_listen_param()

**Comment by spiiroin** [comment]
Provided that caching parse data pointers is ok as-is / fixed -> LGTM

File: src/nci_util.c (hunk 1 above)
Comment (monich) [reply src/nci_util.c:176]: Meh, not worth the trouble IMO. I'm only using first 8 bytes of this SENSF_RES thing, anything other than 16 or 18 would be a violation of NCI 1.0 spec, but it says "Other values are RFU" - who knows maybe some other values are already in use by a newer spec that I don't have.

File: src/nci_util.c (hunk 3 above)
Comment (monich) [reply src/nci_util.c:384]: Yes, the idea is that those "parse" functions don't allocate memory - and they are allowed to use pointers to the memory block which was passed in. The caller is supposed to make sure that those pointers remain valid for as long as necessary.

**Comment by monich** [comment]
Thanks!

**Pull request merged**